src/Pos/Util/UserSecret.hs

```diff
@@ -109,6 +158,7 @@ peekUserSecret path = takeReadLock path $ do
 -- 'writeUserSecretRelease'.
 takeUserSecret :: (MonadIO m) => FilePath -> m UserSecret
 takeUserSecret path = liftIO $ do
+    initializeUserSecret path
```

Comment:
@startling could you elaborate how this is a problem? `writeRaw` always ensures that the key is written with the correct permissions, so the secret should never be exposed.